### ISSUE TITLE
line 63: ansible-playbook: command not found

### ISSUE THREAD

**Issue by joshwardell** [trailofbits/algo#301]
### OS / Environment

MacOS 10.12.4

### Summary of the problem
Trying to install Also on a new google MC2

### Steps to reproduce the behavior

when running .algo in my Mac terminal, when answering questions, it exits:
Please choose the number of your zone. Press enter for default (#8) zone.
[8]: 7

Do you want to enable VPN On Demand when connected to cellular networks?
[y/N]: y

Do you want to enable VPN On Demand when connected to Wi-Fi?
[y/N]: y

Do you want to exclude trusted Wi-Fi networks from using the VPN? (e.g., your home network. Comma-separated value, e.g., HomeNet,OfficeWifi,AlgoWiFi)
: y

Do you want to install a local DNS resolver to block ads while surfing?
[y/N]: y

Do you want each user to have their own account for SSH tunneling?
[y/N]: y

Do you want to apply operating system security enhancements on the server? (warning: replaces your sshd_config)
[y/N]: y

Do you want the VPN to support Windows 10 clients? (requires RSA certificates and key exchange, less secure)
[y/N]: n

Do you want to store the CA key? (required for update-users script, but less secure)
[y/N]: y
./algo: line 63: ansible-playbook: command not found

### Full log

I also had an error in the previous step pip install, see below

pip install virtualenv && virtualenv env && source env/bin/activate && pip install -r requirements.txt
Collecting virtualenv
  Using cached virtualenv-15.1.0-py2.py3-none-any.whl
Installing collected packages: virtualenv
Exception:
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/commands/install.py", line 342, in run
    prefix=options.prefix_path,
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/req/req_set.py", line 784, in install
    **kwargs
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/req/req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/wheel.py", line 345, in move_wheel_files
    clobber(source, lib_dir, True)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/wheel.py", line 323, in clobber
    shutil.copyfile(srcfile, destfile)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 83, in copyfile
    with open(dst, 'wb') as fdst:
IOError: [Errno 13] Permission denied: '/Library/Python/2.7/site-packages/virtualenv.py'

**Comment by dguido** [comment]
You can't find ansible-playbook because you didn't install the requirements. Please install the requirements as described in the readme and try running Algo again.

**Comment by joshwardell** [comment]
But I can't install the requirements.

Here's a full log of what I get in the terminal:

Joshs-MacBook-Pro-15:~ jwardell$ cd /Users/jwardell/Downloads/algo-master 
Joshs-MacBook-Pro-15:algo-master jwardell$ sudo easy_install pip
Password:
Searching for pip
Best match: pip 9.0.1
Processing pip-9.0.1-py2.7.egg
pip 9.0.1 is already the active version in easy-install.pth
Installing pip script to /usr/local/bin
Installing pip2.7 script to /usr/local/bin
Installing pip2 script to /usr/local/bin

Using /Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg
Processing dependencies for pip
Finished processing dependencies for pip
Joshs-MacBook-Pro-15:algo-master jwardell$ pip install virtualenv && virtualenv env && source env/bin/activate && pip install -r requirements.txt
Collecting virtualenv
  Using cached virtualenv-15.1.0-py2.py3-none-any.whl
Installing collected packages: virtualenv
Exception:
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/commands/install.py", line 342, in run
    prefix=options.prefix_path,
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/req/req_set.py", line 784, in install
    **kwargs
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/req/req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/wheel.py", line 345, in move_wheel_files
    clobber(source, lib_dir, True)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/wheel.py", line 323, in clobber
    shutil.copyfile(srcfile, destfile)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 83, in copyfile
    with open(dst, 'wb') as fdst:
IOError: [Errno 13] Permission denied: '/Library/Python/2.7/site-packages/virtualenv.py'
Joshs-MacBook-Pro-15:algo-master jwardell$

**Comment by dguido** [comment]
We just posted some better instructions on the readme (literally, like 3 minutes ago). Can you try blowing away the folder you have with Algo, downloading it again, and trying the new instructions? It might resolve your issue.

**Comment by joshwardell** [comment]
Just my luck...I was eventually successful with the new instructions, with
two fatals that I figured out:

-Must log into Google CE and Enable the server first
-Must rename the json file downloaded from Google CE, removing spaces

Thanks so much for the instant support!

**Comment by dguido** [comment]
Great to hear! If you think the instructions could be clarified somehow, consider submitting a pull request to help other users who might have the same issues. Thanks!

**Comment by inancgumus** [comment]
@dguido The same problem here. I installed the requirements succesfully. Is there an option to debug it?

**Comment by Greenjam94** [comment]
@dguido I'm attempting to install algo on a brand new DO droplet, 512mb ram ubuntu 16.04.2 x64

I'm getting the same error after following all of the steps in the README.md and I just cloned the repo this morning (11am EST)

I attempted to rerun `python -m pip install -r requirements.txt` but it said everything was installed correctly. Only warning was to update pip to 9.0.1 from 8.1.1

Thoughts on how to fix this?

**Comment by gunph1ld** [comment]
@Greenjam94 ensure that you are in virtualenv (check the readme)

**Comment by Greenjam94** [comment]
@gunph1ld You're right, thanks. I was trying to run that from the /algo dir and it wasn't working. By running each command from step 4 by themselves, and cd into algo before running the third, I got it to work.

**Comment by thoatswold** [comment]
I ran into the same problem when trying to install Algo on Windows 10 build 14393, using the instructions in https://github.com/trailofbits/algo/blob/master/docs/WINDOWS.md, which I followed to the letter. Eventually I was able to solve it by updating pip (`python -m pip install --upgrade pip`) **before** installing the dependencies in requirements.txt.

I strongly suggest that this extra step should be added to the Windows instructions at the end, or before cloning the Algo repository.
